src/test/ui/type/ascription/issue-47666.stderr

```diff
@@ -12,5 +32,11 @@ LL |     let _ = Option:Some(vec![0, 1]);
    = note: `#![feature(type_ascription)]` lets you annotate an expression with a type: `<expr>: <type>`
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
 
-error: aborting due to previous error
+error: expected `::`, found `:`
+  --> $DIR/issue-47666.rs:7:16
+   |
+LL |     let _ = Foo:A;
+   |                ^ help: write a path separator instead: `::`
+
```

Comment:
For reference, [this is the current output](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=d6fb7838fd8c2c722d91313f22ac73be):

```
error: expected type, found `2`
 --> src/main.rs:6:19
  |
6 |     let _ = Foo:B(2); //~ ERROR expected type
  |                -  ^ expected type
  |                |
  |                help: maybe write a path separator here: `::`
  |
  = note: `#![feature(type_ascription)]` lets you annotate an expression with a type: `<expr>: <type>`

error: expected one of `!`, `(`, `.`, `::`, `;`, `<`, or `?`, found `{`
 --> src/main.rs:9:19
  |
9 |     let _ = Foo:C { x: 1 }; //~ ERROR expected one of
  |                -  ^ expected one of 7 possible tokens
  |                |
  |                help: maybe write a path separator here: `::`
  |
  = note: `#![feature(type_ascription)]` lets you annotate an expression with a type: `<expr>: <type>`

error: expected type, found reserved keyword `box`
 --> src/main.rs:2:25
  |
2 |     let _ = Option:Some(vec![0, 1]); //~ ERROR expected type, found
  |                   -     ^^^^^^^^^^
  |                   |     |
  |                   |     expected type
  |                   |     in this macro invocation
  |                   |     this macro call doesn't expand to a type
  |                   help: maybe write a path separator here: `::`
  |
  = note: `#![feature(type_ascription)]` lets you annotate an expression with a type: `<expr>: <type>`
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0423]: expected value, found enum `Option`
 --> src/main.rs:2:13
  |
2 |     let _ = Option:Some(vec![0, 1]); //~ ERROR expected type, found
  |             ^^^^^^
  |
help: try using one of the enum's variants
  |
2 |     let _ = std::option::Option::None:Some(vec![0, 1]); //~ ERROR expected type, found
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^
2 |     let _ = std::option::Option::Some:Some(vec![0, 1]); //~ ERROR expected type, found
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^
help: consider importing one of these items instead
  |
1 | use serde::de::Unexpected::Option;
  |
1 | use serde_value::Unexpected::Option;
  |
1 | use serde_value::Value::Option;
  |

error[E0573]: expected type, found variant `Some`
 --> src/main.rs:2:20
  |
2 |     let _ = Option:Some(vec![0, 1]); //~ ERROR expected type, found
  |                    ^^^^^^^^^^^^^^^^ not a type
  |
help: try using the variant's enum
  |
2 |     let _ = Option:core::option::Option; //~ ERROR expected type, found
  |                    ^^^^^^^^^^^^^^^^^^^^
2 |     let _ = Option:futures_core::core_reexport::option::Option; //~ ERROR expected type, found
  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2 |     let _ = Option:futures_util::core_reexport::option::Option; //~ ERROR expected type, found
  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2 |     let _ = Option:lzw::Bits; //~ ERROR expected type, found
  |                    ^^^^^^^^^
    and 5 other candidates
help: maybe you meant to write a path separator here
  |
2 |     let _ = Option::Some(vec![0, 1]); //~ ERROR expected type, found
  |                   ^^

error[E0423]: expected value, found enum `Foo`
 --> src/main.rs:5:13
  |
5 |     let _ = Foo:A; //~ ERROR expected `::`, found `:`
  |             ^^^
  |
help: try using one of the enum's variants
  |
5 |     let _ = Foo::A:A; //~ ERROR expected `::`, found `:`
  |             ^^^^^^
5 |     let _ = Foo::B:A; //~ ERROR expected `::`, found `:`
  |             ^^^^^^
5 |     let _ = Foo::C:A; //~ ERROR expected `::`, found `:`
  |             ^^^^^^

error[E0412]: cannot find type `A` in this scope
 --> src/main.rs:5:17
  |
5 |     let _ = Foo:A; //~ ERROR expected `::`, found `:`
  |                 ^ not found in this scope
  |
help: there is an enum variant `crate::Foo::A`; try using the variant's enum
  |
5 |     let _ = Foo:crate::Foo; //~ ERROR expected `::`, found `:`
  |                 ^^^^^^^^^^
help: maybe you meant to write a path separator here
  |
5 |     let _ = Foo::A; //~ ERROR expected `::`, found `:`
  |                ^^
help: you might be missing a type parameter
  |
4 | fn qux<A>() {
  |       ^^^
```